### ISSUE TITLE
White page in dev mode

### ISSUE THREAD

**Issue by phhoef** [electron-react-boilerplate/electron-react-boilerplate#3146]
## Summary

I am using a pretty old version of this boilerplate.
I was recently forced to upgrade to Electron 11, as it is the first version supporting Apple Silicon.

When the app is packaged, everything is working fine.
But when I do start the app in dev mode, I only see a white/blank page. The dev console is opening, but no errors are shown.
I think the `app.html` and `renderer.dev.js` are loaded, but it feels like the react app is never really started.

Not sure, what to share, as there are no errors.
<details>
<summary>Here are my dependencies:</summary>

```json
"devDependencies": {
    "@babel/core": "^7.12.3",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-proposal-decorators": "^7.12.1",
    "@babel/plugin-proposal-do-expressions": "^7.12.1",
    "@babel/plugin-proposal-export-default-from": "^7.12.1",
    "@babel/plugin-proposal-export-namespace-from": "^7.12.1",
    "@babel/plugin-proposal-function-bind": "^7.12.1",
    "@babel/plugin-proposal-function-sent": "^7.12.1",
    "@babel/plugin-proposal-json-strings": "^7.12.1",
    "@babel/plugin-proposal-logical-assignment-operators": "^7.12.1",
    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.12.1",
    "@babel/plugin-proposal-numeric-separator": "^7.12.1",
    "@babel/plugin-proposal-optional-chaining": "^7.12.1",
    "@babel/plugin-proposal-pipeline-operator": "^7.12.1",
    "@babel/plugin-proposal-throw-expressions": "^7.12.1",
    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
    "@babel/plugin-syntax-import-meta": "^7.10.4",
    "@babel/plugin-transform-react-constant-elements": "^7.12.1",
    "@babel/plugin-transform-react-inline-elements": "^7.12.1",
    "@babel/preset-env": "^7.12.1",
    "@babel/preset-react": "^7.12.1",
    "@babel/preset-typescript": "^7.12.1",
    "@babel/register": "^7.12.1",
    "@types/jest": "^26.0.15",
    "@types/react": "^16.9.55",
    "@types/react-dom": "^16.9.9",
    "@types/react-redux": "^7.1.9",
    "@types/react-router-dom": "^5.1.6",
    "@types/redux-logger": "^3.0.8",
    "@types/serialport": "^8.0.2",
    "babel-core": "7.0.0-bridge.0",
    "babel-eslint": "^10.1.0",
    "babel-jest": "^26.6.1",
    "babel-loader": "^8.1.0",
    "babel-plugin-dev-expression": "^0.2.1",
    "babel-plugin-transform-react-remove-prop-types": "^0.4.21",
    "chalk": "^4.1.0",
    "concurrently": "^5.3.0",
    "connected-react-router": "^6.8.0",
    "cross-env": "^7.0.2",
    "cross-spawn": "^7.0.3",
    "css-loader": "^5.0.0",
    "detect-port": "^1.3.0",
    "electron": "^11.5.0",
    "electron-builder": "^22.9.1",
    "electron-devtools-installer": "^3.1.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.5",
    "enzyme-to-json": "^3.6.1",
    "eslint": "^7.12.1",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-prettier": "^6.15.0",
    "eslint-formatter-pretty": "^4.0.0",
    "eslint-import-resolver-webpack": "^0.13.0",
    "eslint-plugin-compat": "^3.8.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jest": "^24.1.0",
    "eslint-plugin-jsx-a11y": "6.4.1",
    "eslint-plugin-promise": "^4.0.1",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-testcafe": "^0.2.1",
    "fbjs-scripts": "^3.0.0",
    "file-loader": "^6.2.0",
    "husky": "^4.3.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.6.1",
    "lint-staged": "^10.5.1",
    "mini-css-extract-plugin": "^1.2.1",
    "opencollective-postinstall": "^2.0.3",
    "optimize-css-assets-webpack-plugin": "^5.0.4",
    "prettier": "^2.1.2",
    "raw-loader": "^4.0.2",
    "react-test-renderer": "^17.0.1",
    "redux-logger": "^3.0.6",
    "rimraf": "^3.0.2",
    "sass-loader": "^10.0.4",
    "sinon": "^9.2.1",
    "spectron": "^12.0.0",
    "style-loader": "^2.0.0",
    "stylelint": "^13.7.2",
    "stylelint-config-prettier": "^8.0.2",
    "stylelint-config-standard": "^20.0.0",
    "testcafe": "^1.9.4",
    "testcafe-browser-provider-electron": "^0.0.15",
    "testcafe-live": "^0.1.4",
    "testcafe-react-selectors": "^4.0.0",
    "ts-loader": "^8.0.7",
    "typescript": "^4.1.3",
    "url-loader": "^4.1.1",
    "webpack": "^5.3.2",
    "webpack-bundle-analyzer": "^3.9.0",
    "webpack-cli": "^3.3.0",
    "webpack-dev-server": "^3.7.4",
    "webpack-merge": "^5.3.0",
    "yarn": "^1.22.10"
  },
  "dependencies": {
    "@fortawesome/fontawesome-free": "^5.15.1",
    "@types/semver": "^7.3.4",
    "about-window": "^1.13.4",
    "axios": "^0.21.0",
    "dblsqd-electron5": "^0.5.3",
    "dblsqd-sdk": "^0.2.1",
    "devtron": "^1.4.0",
    "electron-builder-squirrel-windows": "^22.9.1",
    "electron-debug": "^3.1.0",
    "electron-is-dev": "^1.2.0",
    "electron-log": "^4.2.4",
    "electron-settings": "^4.0.2",
    "electron-window-state": "^5.0.3",
    "history": "^4.10.1",
    "md5": "^2.3.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-hot-loader": "^4.13.0",
    "react-redux": "^7.2.2",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.2.0",
    "redux": "^4.0.5",
    "redux-thunk": "^2.3.0",
    "rollbar": "^2.24.0",
    "semantic-ui-react": "^2.0.1",
    "semver": "^7.3.2",
    "serialport": "^10.0.1",
    "source-map-support": "^0.5.19",
    "strongly-typed-events": "^1.6.22",
    "usb-detection": "^4.13.0"
  },
```

</details>

Can someone please give me some guidance? I am really lost how I can hands on this problem.

**Comment by kbrisso** [comment]
Is there a stack trace in the cmd window?

**Comment by phhoef** [comment]
No, there is neither an error/stack trance in the cmd window nor in the dev console of chrome's dev tools.
That's why I am running out of ideas, what the problem is. I've no starting point to track the problem down ...

**Comment by westside** [comment]
I have a same issue here on my mac. Any suggestions here?

**Comment by phhoef** [comment]
I found the issue a few days ago.
In my case it was a native dependency, not compatible with M1 Mac. For some reason I removed the `electron-builder install-app-dep` as `postinstall` script. So, `yarn` was able to finish the install process. Though, the native dependency was not compiled for my arch. Then, when Electron started I got only the blank page without any errors.

I removed the native dependency and built a workaround to achieve the same functionality.
Hope this helps ...
